kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -152,6 +161,170 @@ void prometheusMetricsWithCommonTags(KafkaCluster cluster) {
+
+        // the downstream messages and decoded messages is not yet differentiated by ApiKey
+        final UUID configInstance = UUID.randomUUID();
+
+        NamedFilterDefinition namedFilterDefinition = new NamedFilterDefinitionBuilder("filter",
+                CreateTopicRequest.class.getName()).withConfig("configInstanceId", configInstance).build();
+
+        var config = proxy(cluster)
+                .addToFilterDefinitions(namedFilterDefinition)
+                .addToDefaultFilters(namedFilterDefinition.name())
+                .withNewManagement()
+                .withNewEndpoints()
+                .withNewPrometheus()
+                .endPrometheus()
+                .endEndpoints()
+                .endManagement();
+
+        try (var tester = kroxyliciousTester(config);
+                var managementClient = tester.getManagementClient();
+                var producer = tester.producer()) {
+
+            var metricList = managementClient.scrapeMetrics();
+
+            var inboundDownstreamMessagesMetricsValue = getMetricsValue(metricList, "kroxylicious_inbound_downstream_messages_total", null);
+
+            var inboundDownstreamDecodedMessagesMetricsValue = getMetricsValue(metricList, "kroxylicious_inbound_downstream_decoded_messages_total", null);
+
+            producer.send(new ProducerRecord<>(topic.name(), "my-key", "hello-world")).get();
+
+            // updated metrics after some message were produced
+            var updatedMetricsList = managementClient.scrapeMetrics();
+
+            var updatedInboundDownstreamMessagesMetricsValue = getMetricsValue(updatedMetricsList, "kroxylicious_inbound_downstream_messages_total", null);
+            var updatedInboundDownstreamDecodedMessagesMetricsValue = getMetricsValue(updatedMetricsList, "kroxylicious_inbound_downstream_decoded_messages_total", null);
+
+            assertThat(updatedInboundDownstreamMessagesMetricsValue).isGreaterThan(inboundDownstreamMessagesMetricsValue);
+            assertThat(updatedInboundDownstreamDecodedMessagesMetricsValue).isGreaterThan(inboundDownstreamDecodedMessagesMetricsValue);
+        }
+    }
+
+    @Test
+    void shouldIncrementConnectionMetrics(KafkaCluster cluster, Topic topic) throws ExecutionException, InterruptedException {
+        var config = proxy(cluster)
+                .addToFilterDefinitions()
+                .addToDefaultFilters()
+                .withNewManagement()
+                .withNewEndpoints()
+                .withNewPrometheus()
+                .endPrometheus()
+                .endEndpoints()
+                .endManagement();
+
+        try (var tester = kroxyliciousTester(config);
+                var managementClient = tester.getManagementClient()) {
+
+            var metricsList = managementClient.scrapeMetrics();
+
+            assertThat(checkMetricsExistOrNot(metricsList, "kroxylicious_downstream_connections_total", null)).isEqualTo(false);
```

Comment:
Thats not really how I expect to use `assertThat` I guess its because your trying to re-use `checkMetricExistsOrNot`.

I'd have expected something more like

```suggestion
            assertThat(metricsList)
                    .hasSizeGreaterThan(0)
                    .anySatisfy(simpleMetric -> {
                        assertThat(simpleMetric.name()).isEqualTo("kroxylicious_downstream_connections_total");
                        assertThat(simpleMetric.labels()).containsValue(apiKey.toString()); 
                    });
```

We could also extract a method to wrap that assertion, or write a custom Metrics assertion....

Our use of assertJ has a evolved a lot since I wrote the original MetricsIT. So its a poor reference, sorry about that.

Reply:
Hi Sam, I am just wondering why we do this `assertThat(simpleMetric.labels()).containsValue(apiKey.toString()); `? Wouldn't this mean that it would check all the metrics to find any metric with the provided api key

Reply:
Another question would be `what if the metrics doesn't exist` ? We can assert this only if the metrics exist, if the metrics doesn't exist then we will get assertion error

Reply:
Yes your right both assertions have to pass for the value to satisfy and this only one with the right name and label will pass.


If we want to assert that there are no metrics there is a version for the inverse. I don't remember the correct name right now.

Reply:
Added the suggestions now